src/Files/Filesystem/StorageItems/SystemStorageFile.cs

```diff
@@ -77,22 +77,48 @@ public override IAsyncAction CopyAndReplaceAsync(IStorageFile fileToReplace)
 
         public override IAsyncAction MoveAsync(IStorageFolder destinationFolder)
         {
-            return File.MoveAsync(destinationFolder);
+            return MoveAsync(destinationFolder, Name, NameCollisionOption.FailIfExists);
         }
 
         public override IAsyncAction MoveAsync(IStorageFolder destinationFolder, string desiredNewName)
         {
-            return File.MoveAsync(destinationFolder, desiredNewName);
+            return MoveAsync(destinationFolder, desiredNewName, NameCollisionOption.FailIfExists);
         }
 
         public override IAsyncAction MoveAsync(IStorageFolder destinationFolder, string desiredNewName, NameCollisionOption option)
         {
-            return File.MoveAsync(destinationFolder, desiredNewName, option);
+            return AsyncInfo.Run(async (cancellationToken) =>
```

Comment:
Why not Task.Run?

Reply:
Need to return an AsyncAction. That's how the IStorageItem interface is.